### ISSUE TITLE
false positive truthy failure on travis files

### ISSUE THREAD

**Issue by ssbarnea** [adrienverge/yamllint#158]
It seems that yamllint chokes complaining about not using truthy values for travis **keys**.

Just to be clear: this is about **keys not values with name "on".**. When used as a key, on have a total different sense than the "on" as a value, which I find acceptable to trigger a warning.

**Comment by ssbarnea** [comment]
Thanks for the update, now we need to find a way to fix this because having it documented does not make the user experience better. So far the only place I found to be conflicting is travis.yaml and to be honest I cannot find any reasom for blaming its syntax here.

Maybe we need an option to look for truthy only for leaf-keys or to disable it for travis files. I am open to suggestions. 

Fixing this is not an urgency as I disabled the truthy feature completely for the moment on that small project. Still, we should look for a way to address this in the default tool behaviour, so the user experience will be good and he will not encounter this false-positive.

**Comment by ssbarnea** [comment]
My mistake to not give an example, https://github.com/pycontribs/tendo/blob/master/.travis.yml#L78

We are lucky the the "on" condition (which is similar to `when` on ansible) is not very common and is not present in default travis templates. Still, is a core feature which is very old and established so I doubt they will change the file format to pass yamllint-ing.

I don't know what is the right approach to take here but I know how to describe the desired user experience: when user enables yamllint they should not get a false-positive on a travis.yaml file.

**Comment by adrienverge** [comment]
What about a new option for the `truthy` rule? Something like `check-keys: true | false` or similar?

@pdericson any thoughts?

**Comment by jayvdb** [comment]
What about a special case for `.travis.yml` , or having schemas which define key names which could then be whitelisted if they appear in the correct place in the yaml.

**Comment by jb-sean** [comment]
This issue is not limited to `.travis.yml` configs. The new GitHub Actions yaml spec has a required `on` key: https://help.github.com/en/articles/workflow-syntax-for-github-actions#on
E.g.,
```yaml
name: Pull request based on master
on:
  pull_request:
    branches:
      - master
# ...
```
yamllint is throwing `2:1       warning  truthy value should be one of [false, true]  (truthy)`.

**Comment by adrienverge** [comment]
Hi,

While waiting for someone to contribute a new `check-keys` option or something equivalent, an option is to disable the `truthy` rule in yamllint config:
```yaml
---
extends: default
rules:
  truthy: disable
```
... or allow `on`:
```yaml
---
extends: default
rules:
  truthy:
    allowed-values: ['true', 'false', 'on']
```

**Comment by jameswilliams1** [comment]
An easy workaround here, add:
```yml
# yamllint disable-line rule:truthy
```
to any ci lines that use 'on' which avoids having to fully disable truthy (personally I like this rule for Ansible roles and don't want to disable it due to a bug).

**Comment by oerp-odoo** [comment]
Is this issue fixed?

Im still getting that warning on `github` actions. Should I just use `# yamllint disable-line rule:truthy` to disable it?

**Comment by brettinternet** [comment]
To fix this just add this to your `.yamllint`

```yml
rules:
  truthy:
    allowed-values: ['true', 'false']
    check-keys: false
```
